bindings/matrix-sdk-ffi/src/room.rs

```diff
@@ -656,11 +656,11 @@ impl Room {
     /// Mark a room as read, by attaching a read receipt on the latest event.
     ///
     /// Note: this does NOT unset the unread flag; it's the caller's
-    /// responsibility to do so, if needs be.
+    /// responsibility to do so, if need be.
     pub async fn mark_as_read(&self, receipt_type: ReceiptType) -> Result<(), ClientError> {
-        let timeline = self.timeline().await?;
+        let timeline = TimelineBuilder::new(&self.inner).build().await?;
```

Comment:
Could this be: if there's a timeline created for this room, then use it; otherwise create a temporary one (like you're doing here)?

Reply:
This was actually a miss on my part, the ffi room should no longer hold on to a timeline instance, which I've taken care of in the last commit.